src/neo/Plugins/ITraceDebugPlugin.cs

```diff
@@ -0,0 +1,10 @@
+using Neo.Network.P2P.Payloads;
+
+namespace Neo.Plugins
+{
+    public interface ITraceDebugPlugin
+    {
+        bool ShouldTrace(Header header, Transaction tx);
```

Comment:
Is not enough with `GetSink` ? it's needed `ShouldTrace`?

Reply:
The idea here is that you don't likely want to trace every tx. For example, if I deploy a contract to mainnet, I can also deploy a mainnet node with the debugger plugin to capture every run of my contract for later analysis if need be. But I don't want to capture every tx - just mine. This API allows me to cleanly separate the logic that decides if a given tx should be traced vs. the logic that does the tracing.

Reply:
But this logic can be done in the plugin, if you don't want to trace this TX, just ignore it, isn't it?

Reply:
Yes, it could be done other ways. I feel this approach provides more flexibility in case we want to support multiple tracers for a single transaction.

Reply:
What do you think @erikzhang ?